src/node_worker.cc

```diff
@@ -58,6 +58,40 @@ void WaitForWorkerInspectorToStop(Environment* child) {
 
 }  // anonymous namespace
 
+void LoopStopper::Install(uv_loop_t* stopper,
+                          void* data,
+                          uv_async_cb target,
+                          bool unref) {
+  async_ = new uv_async_t;
+  if (data != nullptr) async_->data = data;
+  CHECK_EQ(uv_async_init(stopper, async_, target), 0);
+  if (unref) uv_unref(reinterpret_cast<uv_handle_t*>(async_));
+}
+
+void LoopStopper::Uninstall() {
+  Mutex::ScopedLock lock(mutex_);
+  if (async_ != nullptr) {
+    uv_close(reinterpret_cast<uv_handle_t*>(async_),
```

Comment:
Can you keep using `env->CloseHandle()` here? It’s always true so far, but it’s not part of the libuv API contract that closing an `async` handle finishes within the same event loop iteration (see e.g. https://github.com/nodejs/node/pull/26088)

Reply:
@addaleax - thanks, done. In fact I was worried about the async nature of `uv_close` but was unable to figure out what to do with that.

Changed some of the class semantics to accommodate this.